test/e2e/service/nlb_ip_target_test.go

```diff
@@ -309,15 +317,19 @@ var _ = Describe("k8s service reconciled by the aws load balancer", func() {
 			lbName string
 		)
 		BeforeEach(func() {
+			annotation := map[string]string{
+				"service.beta.kubernetes.io/aws-load-balancer-name":   lbName,
+				"service.beta.kubernetes.io/aws-load-balancer-type":   "nlb-ip",
+				"service.beta.kubernetes.io/aws-load-balancer-scheme": "internet-facing",
+			}
+			if tf.Options.IPFamily == "IPv6" {
+				annotation["service.beta.kubernetes.io/aws-load-balancer-ip-address-type"] = "dualstack"
+			}
 			lbName = utils.RandomDNS1123Label(20)
```

Comment:
You need to move this assignment before you create the annotation map. Otherwise, the lb annotation value will be empty - and the tests fail :)